CONTRIBUTING.md

```diff
@@ -97,28 +97,21 @@ validates all configurations options when provided before generating
 its config file.
 
 The `PrometheusCharm` object interacts with its scrape targets using a
-[charm library](lib/charms/prometheus_k8s/v1/prometheus.py). Using this
-library requires that Prometheus informs its "Consumers" (scrape targets)
-of the actual Prometheus version that was deployed. In order to determine
-this version at runtime `PrometheuCharm` uses the
-[`Prometheus`](src/prometheus_server.py) object. The `Prometheus`
-object provides an interface to a running Prometheus instance. This
-interface is limited to only those aspects of Prometheus required by
-this charm.
+[charm library](lib/charms/prometheus_k8s/v1/prometheus.py).
 
 ### Library Details
 
-The Prometheus charm library exposes a provider and consumer object -
-`PrometheusProvider` and `PrometheusConsumer` along with the custom
+The Prometheus charm library exposes a consumer and provider object -
+`MetricsConsumer` and `MetricsProvider` along with the custom
```

Comment:
I thought we'd call them "MetricsEndpoint{Provider|Consumer}"

Reply:
The relation is called `metrics-endpoint`. Since these classes derive form `ProviderBase` and `ConsumerBase` they are named `MetricsProvider` and `MetricsConsmer` .

Reply:
It is metrics that is being provided and consumed. Isn't it ?

Reply:
I think I understand what you went for, but if we wanna go philosophical, it is not metrics that are being provided and consumed: it is metrics endpoints. The metrics are the outcome of consuming the metrics endpoint, which happens on a completely different level, outside of Juju. If the actual metrics data were somehow shared over relations (horrible, horrible idea that nobody should even consider), then it would be `MetricsProvider` and `MetricsConsumer`.